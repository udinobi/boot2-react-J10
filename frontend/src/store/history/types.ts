import { Action } from "redux"

import { Location, LocationSelectedAction } from "../location/city/types"

export interface HistoryLocation {
    location: Location
}

export interface HistoryState {
    locations: Location[]
}

export enum HistoryActionType {
    LOCATION_RELOAD = "LOCATION_RELOAD",
    LOCATION_REMOVE = "LOCATION_REMOVE"
}

export interface LocationReloadAction extends Action {
    type: HistoryActionType.LOCATION_RELOAD
    payload: {
        location: Location
    }
}

export interface LocationRemoveAction extends Action {
    type: HistoryActionType.LOCATION_REMOVE
    payload: {
        location: Location
    }
}

export type HistoryActions =
    LocationRemoveAction | LocationSelectedAction
