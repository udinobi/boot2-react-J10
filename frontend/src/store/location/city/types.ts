import { Action } from "redux"

import { Option } from "tsoption"

import { Country, CountrySelectedAction } from "../country/types"

export interface Coordinates {
    lat: string
    lon: string
}

export interface Location {
    geoId: number
    name: string
    coord: Coordinates
    country: Country
}

export interface CityState {
    country: Option<Country>,
    suggestions: Location[]
}

export enum CityActionType {
    LOCATION_SELECTED = "LOCATION_SELECTED",
    SUGGESTIONS_FAILURE = "SUGGESTIONS_FAILURE",
    SUGGESTIONS_LOOKUP = "SUGGESTIONS_LOOKUP",
    SUGGESTIONS_RESET = "SUGGESTIONS_RESET",
    SUGGESTIONS_RETRIEVED = "SUGGESTIONS_RETRIEVED"
}

export interface LocationSelectedAction extends Action {
    type: CityActionType.LOCATION_SELECTED
    payload: {
        location: Location
    }
}

export interface SuggestionsFailureAction extends Action {
    type: CityActionType.SUGGESTIONS_FAILURE
}

export interface SuggestionsLookupAction extends Action {
    type: CityActionType.SUGGESTIONS_LOOKUP,
    payload: {
        country: Country,
        locationTerm: string
    }
}

export interface SuggestionsResetAction extends Action {
    type: CityActionType.SUGGESTIONS_RESET
}

export interface SuggestionsRetrievedAction extends Action {
    type: CityActionType.SUGGESTIONS_RETRIEVED
    payload: {
        country: Country
        suggestions: Location[]
    }
}

export type CityActions =
    CountrySelectedAction |
    SuggestionsFailureAction | SuggestionsLookupAction |
    SuggestionsResetAction | SuggestionsRetrievedAction
