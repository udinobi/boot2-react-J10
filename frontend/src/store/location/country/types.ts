import { Action } from "redux"

export interface Country {
    code: string
    name: string
}

export interface CountryState {
    countries: Country[]
}

export enum CountryActionType {
    COUNTRIES_FAILURE = "COUNTRIES_FAILURE",
    COUNTRIES_LOADING = "COUNTRIES_LOADING",
    COUNTRIES_RETRIEVED = "COUNTRIES_RETRIEVED",
    COUNTRY_SELECTED = "COUNTRY_SELECTED"
}

export interface CountriesFailureAction extends Action {
    type: CountryActionType.COUNTRIES_FAILURE
}

export interface CountriesLoadingAction extends Action {
    type: CountryActionType.COUNTRIES_LOADING
}

export interface CountriesRetrievedAction extends Action {
    type: CountryActionType.COUNTRIES_RETRIEVED
    payload: {
        countries: Country[]
    }
}

export interface CountrySelectedAction extends Action {
    type: CountryActionType.COUNTRY_SELECTED
    payload: {
        country: Country
    }
}

export type CountryActions =
    CountriesFailureAction | CountriesLoadingAction | CountriesRetrievedAction
