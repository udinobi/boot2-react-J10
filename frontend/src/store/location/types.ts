import { Action } from "redux"

import { Option } from "tsoption"

export interface Coordinates {
    lat: string
    lon: string
}

export interface Country {
    code: string
    name: string
}

export interface Location {
    geoId: number
    name: string
    coord: Coordinates
    country: Country
}

// Definitions related to the Country element ------------------------------------------------------

export interface CountryState {
    countries: Country[]
}

export enum CountryActionType {
    COUNTRIES_FAILURE = "COUNTRIES_FAILURE",
    COUNTRIES_LOADING = "COUNTRIES_LOADING",
    COUNTRIES_RETRIEVED = "COUNTRIES_RETRIEVED",
    COUNTRY_SELECTED = "COUNTRY_SELECTED"
}

export interface CountriesFailureAction extends Action {
    type: CountryActionType.COUNTRIES_FAILURE
}

export interface CountriesLoadingAction extends Action {
    type: CountryActionType.COUNTRIES_LOADING
}

export interface CountriesRetrievedAction extends Action {
    type: CountryActionType.COUNTRIES_RETRIEVED
    payload: {
        countries: Country[]
    }
}

export interface CountrySelectedAction extends Action {
    type: CountryActionType.COUNTRY_SELECTED
    payload: {
        country: Country
    }
}

export type CountryActions =
    CountriesFailureAction | CountriesLoadingAction | CountriesRetrievedAction

// Definitions related to the City element ---------------------------------------------------------

export interface CityState {
    country: Option<Country>,
    suggestions: Location[]
}

export enum CityActionType {
    LOCATION_SELECTED = "LOCATION_SELECTED",
    SUGGESTIONS_FAILURE = "SUGGESTIONS_FAILURE",
    SUGGESTIONS_LOOKUP = "SUGGESTIONS_LOOKUP",
    SUGGESTIONS_RESET = "SUGGESTIONS_RESET",
    SUGGESTIONS_RETRIEVED = "SUGGESTIONS_RETRIEVED"
}

export interface LocationSelectedAction extends Action {
    type: CityActionType.LOCATION_SELECTED
    payload: {
        location: Location
    }
}

export interface SuggestionsFailureAction extends Action {
    type: CityActionType.SUGGESTIONS_FAILURE
}

export interface SuggestionsLookupAction extends Action {
    type: CityActionType.SUGGESTIONS_LOOKUP
}

export interface SuggestionsResetAction extends Action {
    type: CityActionType.SUGGESTIONS_RESET
}

export interface SuggestionsRetrievedAction extends Action {
    type: CityActionType.SUGGESTIONS_RETRIEVED
    payload: {
        suggestions: Location[]
    }
}

export type CityActions =
    CountrySelectedAction | LocationSelectedAction | SuggestionsFailureAction |
    SuggestionsLookupAction | SuggestionsResetAction | SuggestionsRetrievedAction
