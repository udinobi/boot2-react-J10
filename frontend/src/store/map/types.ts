import { Option } from "tsoption"

import { LocationReloadAction } from "../history/types"

import { Location, LocationSelectedAction } from "../location/city/types"

export interface MapState {
    location: Option<Location>
}

export type MapActions = LocationReloadAction | LocationSelectedAction
