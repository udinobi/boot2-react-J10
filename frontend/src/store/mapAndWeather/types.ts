import { Option } from "ts-option"

import { LocationReloadAction } from "../history/types"

import { Location, LocationSelectedAction } from "../location/city/types"

export interface MapAndWeatherState {
    location: Option<Location>
}

export type MapAndWeatherActions = LocationReloadAction | LocationSelectedAction
