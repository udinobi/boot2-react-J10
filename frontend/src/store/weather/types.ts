import { Action } from "redux"

import { Option } from "ts-option"

import { Location } from "../location/city/types"

import { MapAndWeatherActions, MapAndWeatherState } from "../mapAndWeather/types"

export interface Weather {
    description: string
    icon: string
    id: number
    main: string
}

export interface WeatherData {
    base?: string
    clouds?: {
        all: number
    }
    cod?: number
    coord: {
        lat: number
        lon: number
    }
    dt?: number
    id?: number
    main: {
        grnd_level: number
        humidity: number
        pressure: number
        sea_level: number
        temp: number
        temp_max: number
        temp_min: number
    }
    name?: string
    rain?: {
        h3: string
        val: number
    }
    snow?: {
        h3: string
        val: number
    }
    sys: {
        sunrise: number
        sunset: number
    }
    visibility?: number
    weather: Weather[]
    wind?: {
        deg: number
        speed: number
    }
}

export interface WeatherState extends MapAndWeatherState {
    askingWeatherDataToOWM: boolean
    lastUpdate: Option<Date>
    weatherData: Option<WeatherData>
}

export enum WeatherActionType {
    WEATHER_DATA_FAILURE = "WEATHER_DATA_FAILURE",
    WEATHER_DATA_RETRIEVE = "WEATHER_DATA_RETRIEVE",
    WEATHER_DATA_RETRIEVED = "WEATHER_DATA_RETRIEVED"
}

export interface WeatherDataFailureAction extends Action {
    type: WeatherActionType.WEATHER_DATA_FAILURE
}

export interface WeatherDataRetrieveAction extends Action {
    type: WeatherActionType.WEATHER_DATA_RETRIEVE,
    payload: {
        location: Location
        owmApiKey: string
    }
}

export interface WeatherDataRetrievedAction extends Action {
    type: WeatherActionType.WEATHER_DATA_RETRIEVED
    payload: {
        location: Location
        weatherData: WeatherData
    }
}

export type WeatherActions =
    MapAndWeatherActions | WeatherDataFailureAction | WeatherDataRetrievedAction
