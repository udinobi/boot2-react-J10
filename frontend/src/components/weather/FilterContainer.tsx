import React from "react"

import WeatherContainer from "./WeatherContainer"

import CenteredDiv from "../lib/CenteredDiv"

import { WeatherState } from "../../store/weather/types"

export default (props: WeatherState) => 
    props.location.map(location =>
        props.weatherData
            .map(weatherData => <WeatherContainer
                /* tslint:disable:jsx-key */
                    location={location}
                    lastUpdate={props.lastUpdate}
                    weatherData={weatherData}
                />
            )
            .getOrElse(<CenteredDiv>No weather information available!</CenteredDiv>)
    )
    .getOrElse(<div />)
