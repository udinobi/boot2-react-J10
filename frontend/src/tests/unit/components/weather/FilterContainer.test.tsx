import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import React from "react"

import { none } from "ts-option"

import FilterContainer from "../../../../components/weather/FilterContainer"

import { WeatherState } from "../../../../store/weather/types"

describe("--> Snapshot for Components > weather > 'FilterContainer'", () => {
    let props: WeatherState

    beforeEach(() => {
        props = {
            askingWeatherDataToOWM: false,
            lastUpdate: none,
            location: none,
            weatherData: none
        }
    })

    it("matches", () => {
        const wrapper = shallow(<FilterContainer {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
