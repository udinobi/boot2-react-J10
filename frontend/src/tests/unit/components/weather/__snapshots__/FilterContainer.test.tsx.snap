// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--> Snapshot for Components > weather > 'FilterContainer' matches 1`] = `<div />`;
